cpp/src/io/utilities/wrapper_utils.hpp

```diff
@@ -146,3 +146,39 @@ struct rmm_deleter {
 };
 template <typename T>
 using device_ptr = std::unique_ptr<T, rmm_deleter<T>>;
+
+// DOXY
+template <typename T>
+class rmm_device_buffer {
+	T* d_data = nullptr;
+
+public:
+	rmm_device_buffer() noexcept = default;
+	rmm_device_buffer(size_t cnt) {
```

Comment:
This should accept a stream that defaults to 0.

Reply:
I'll make that change.
Do you also propose we use the same stream when calling rmm_free? I have a concern about using a stream there, because we don't have a guarantee that the stream will still exist when the object is destroyed. What are you thoughts on that?

Reply:
Yeah, the `rmm::device_buffer` should maintain state for the stream it was allocated with, and use the same stream for `free`ing.

If a user destroys a stream out from under a `rmm::device_buffer`, I'd file that one under "Undefined Behavior".

With the direction I'd like to go in cuDF, we'll maintain a global set pool of streams, and the streams in that pool wouldn't be destroyed until the end of the program. 

We could also consider passing in the stream as a `shared_ptr` so the device buffer takes shared ownership to ensure it's not destroyed until the device buffer is done using it.